### DISCUSSION TITLE
printing the solver residual values

### DISCUSSION THREAD

**Discussion by Thiyaga2025** [pybamm-team/PyBaMM#4313] [Q&A] [answered]
hi all,

is there an option in pybamm to print the solver residual values while running the simulation. I did see the below command in solver setup, but it doesn't seem to give residual values. 

solver = pybamm.CasadiSolver(mode="safe", extra_options_setup={"print_stats":True})

thanks for your time.

**Comment by martinjrobins** [comment, marked as answer]
no, there isn't an option like this. When are you interested in the residual values? Generally the residual values are calculated in the C code of the solver and wouldn't be accessible unless you felt like editing this code. However, if you are interested in the residual values at the output timesteps, then you can probably calculate these from python using the information stored in the `pybamm.Solution` class

**Reply by Thiyaga2025** [reply]
hi @martinjrobins , thanks a lot for the response. i just wanted to check the residual values as was facing divergence issues with degradation modelling with DFN solver(but SPM/SPMe works).. will check pybamm.Solution class.
